### PR TITLE
Some test fixes

### PR DESCRIPTION
**Pull request by martinpitt**
Again broken out from #16942, to test them in isolation.

In particular the TF fedora-34 fix affects main already, so this should land ASAP.

### CODE REVIEW

**Review by KKoukiou — APPROVED** [review]

**Comment by martinpitt** [comment]
There is still this [pesky TestSession.testBasic failure](https://logs.cockpit-project.org/logs/pull-16962-20220209-212559-541853ca-fedora-coreos/log.html#126-2). I can reproduce it locally on fedora-coreos after a few retries. I think @allisonkarlitskaya saw it, too.

I can reproduce it locally after a few retries:

    for i in `seq 20`; do TEST_OS=fedora-coreos test/verify/check-session TestSession.testBasic $RUNC -stv; done

After it happens, the "admin" session does not go away at all any more, even after several minutes.
```
[root@fedora-coreos-127-0-0-2-2201 ~]# loginctl session-status 54
54 - admin (1001)
           Since: Thu 2022-02-10 04:42:37 UTC; 6min ago
          Leader: 8660
          Remote: 127.0.0.1
         Service: sshd; type tty; class user
           State: closing
            Unit: session-54.scope
                  └─8687 ssh-agent -s -D

Feb 10 04:42:37 fedora-coreos-127-0-0-2-2201 systemd[1]: Started Session 54 of User admin.
Feb 10 04:42:37 fedora-coreos-127-0-0-2-2201 sudo[8692]:    admin : PWD=/run/user/1001 ; USER=root ; COMMAND=/usr/bin/cockpit-bridge --privileged
Feb 10 04:42:37 fedora-coreos-127-0-0-2-2201 sudo[8692]: pam_unix(sudo:session): session opened for user root(uid=0) by (uid=1001)
Feb 10 04:42:37 fedora-coreos-127-0-0-2-2201 cockpit/ssh[8675]: invalid or unusable locale: en_US.UTF-8
Feb 10 04:42:56 fedora-coreos-127-0-0-2-2201 sudo[8692]: pam_unix(sudo:session): session closed for user root
Feb 10 04:42:56 fedora-coreos-127-0-0-2-2201 sshd[8674]: Received disconnect from 127.0.0.1 port 47860:11: Bye Bye
Feb 10 04:42:56 fedora-coreos-127-0-0-2-2201 sshd[8674]: Disconnected from user admin 127.0.0.1 port 47860
Feb 10 04:42:56 fedora-coreos-127-0-0-2-2201 sshd[8660]: pam_unix(sshd:session): session closed for user admin
```

So it's that leftover ssh agent which keeps it open. Possibly because the session was never properly terminated? The code says

```py
       # Kill session from the inside
        b.switch_to_top()
        b.wait_visible("#content")
        b.wait_visible("#hosts-sel")
```
But there's no actual killing going on here.. It looks like this test got damaged at some point?

**Comment by martinpitt** [comment]
I added a TestSession fix, let's see how this fares. Locally it works fine now.

**Comment by martinpitt** [comment]
[NOOOOOOOoooooooo....](https://logs.cockpit-project.org/logs/pull-16962-20220210-052109-a03ef07e-fedora-coreos/log.html#303) Back to the drawing board..

Apparently there is a previous closing session around from a previous test run.

File: test/verify/check-session [hunk 1]
```diff
@@ -71,9 +71,8 @@ class TestSession(MachineCase):
         wait_session(True)
 
         # Kill session from the inside
-        b.switch_to_top()
-        b.wait_visible("#content")
-        b.wait_visible("#hosts-sel")
+        b.logout()
```
Comment (allisonkarlitskaya) [inline test/verify/check-session:74]: It seems like the original intention of the test is something else than what you end up testing here, no?

Comment (martinpitt) [reply test/verify/check-session:74]: Maybe, but what would it be? The users page has a "terminate session" button, but you can't use that for the currently running session. Maybe it was relying on the test to automatically clean up sessions, but as there were no assertions at the end, the whole paragraph would be pointless. I could alternatively just completely remove it, but making sure that a logout properly cleans up the session seems to make sense?

**Comment by allisonkarlitskaya** [comment]
let's land this now.  we can better argue about the details later.  it might be nice to be able to land other PRs today 😅

**Review by allisonkarlitskaya — APPROVED** [review]
👍

File: test/verify/check-session (hunk 1 above)
Comment (allisonkarlitskaya) [reply test/verify/check-session:74]: Ya.  I actually agree with this assessment on the second read.  I have no idea what it was trying to accomplish before.

**Comment by martinpitt** [comment]
It's still suspicious that [so many tests are flaking](https://logs-https-frontdoor.apps.ocp.ci.centos.org/logs/pull-16962-20220210-080844-c18704c6-fedora-coreos/log.html) on the initial login. That does not seem to happen in yesterday's PRs, fcos is fairly clean there.

I can even reproduce that locally, and it even seems to happen *at the very first test* right after booting, i.e. it can't just be related to the "terminate session" cleanup fix. In fact none of the changes here affect the container startup.

```
# ----------------------------------------------------------------------
# testBasic (__main__.TestSession)
Starting machine1 fedora-coreos-127.0.0.2-2201
Assuming machine is already running
+ ! test -f /run/nologin && cat /proc/sys/kernel/random/boot_id
1abef374-30b2-4d44-b780-6f2d907746c1
+ journalctl --show-cursor -n0 -o cat | sed 's/^.*cursor: *//'
s=f35fbca747ae41e6ae4efdc663ac30af;i=1629;b=1abef37430b24d44b7806f2d907746c1;m=5d214ae;t=5d7a612dc52ce;x=41d054f8431962e2
-> Emulation.setDeviceMetricsOverride({"width": 1920, "height": 1200, "deviceScaleFactor": 0, "mobile": false})
Started /usr/lib64/chromium-browser/headless_shell (pid 35232) on port 9307

DevTools listening on ws://127.0.0.1:9307/devtools/browser/2e5efa0b-c197-4d33-a115-a712144ca7e3
[0210/095244.570959:ERROR:gpu_init.cc(453)] Passthrough is not supported, GL is swiftshader, ANGLE is 
CDP: executionContextCreated {"context":{"id":1,"origin":"://","name":"","uniqueId":"-4823293432612784198.-5009512227711019971","auxData":{"isDefault":true,"type":"default","frameId":"B7D4FA269922BAFB9A47BB4A8C144B97"}}}
<- {}
+ cat > '/etc/cockpit/cockpit.conf'
+ logger -p user.info 'COCKPITTEST: start TestSession.testBasic'
+ if test -e /etc/passwd; then echo yes; fi
yes
+ mkdir -p /var/lib/cockpittest && cp -a /etc/passwd /var/lib/cockpittest/_etc_passwd
+ if test -e /etc/group; then echo yes; fi
yes
+ mkdir -p /var/lib/cockpittest && cp -a /etc/group /var/lib/cockpittest/_etc_group
+ if test -e /etc/shadow; then echo yes; fi
yes
+ mkdir -p /var/lib/cockpittest && cp -a /etc/shadow /var/lib/cockpittest/_etc_shadow
+ if test -e /etc/gshadow; then echo yes; fi
yes
+ mkdir -p /var/lib/cockpittest && cp -a /etc/gshadow /var/lib/cockpittest/_etc_gshadow
+ if test -e /etc/subuid; then echo yes; fi
yes
+ mkdir -p /var/lib/cockpittest && cp -a /etc/subuid /var/lib/cockpittest/_etc_subuid
+ if test -e /etc/subgid; then echo yes; fi
yes
+ mkdir -p /var/lib/cockpittest && cp -a /etc/subgid /var/lib/cockpittest/_etc_subgid
+ ls /home
admin
core
+ systemctl try-restart sssd || true
+ loginctl list-sessions
SESSION UID USER SEAT TTY
      9   0 root      

1 sessions listed.
+ podman ps --quiet --all --filter name=ws
3e88f361f03e
+ echo 3e88f361f03e | xargs --no-run-if-empty podman rm -f
3e88f361f03ef11c8aed77a773a66a7dc3048e943d6ad685674be16df6cb8791
+ podman container runlabel RUN cockpit/ws -- --no-tls
185bcfd2c53e86bd88e1db47a3c3f787188aba9d7f5ac124031ba2a6025a47ed
+ curl --insecure --silent --connect-timeout 2 --max-time 3 http://localhost:9090
-> switch to frame None
-> Page.navigate({"url": "http://127.0.0.2:9091/system"})
<- {'frameId': 'B7D4FA269922BAFB9A47BB4A8C144B97', 'loaderId': '547346355BADC152FF335B9DBC407481'}
-> expect_load
CDP: frameNavigated {"frame":{"id":"B7D4FA269922BAFB9A47BB4A8C144B97","loaderId":"547346355BADC152FF335B9DBC407481","url":"http://127.0.0.2:9091/system","domainAndRegistry":"","securityOrigin":"http://127.0.0.2:9091","mimeType":"text/html","adFrameStatus":{"adFrameType":"none"},"secureContextType":"SecureLocalhost","crossOriginIsolatedContextType":"NotIsolated","gatedAPIFeatures":[]},"type":"Navigation"}
CDP: executionContextCreated {"context":{"id":2,"origin":"http://127.0.0.2:9091","name":"","uniqueId":"6147362160004783953.-3708236277066869593","auxData":{"isDefault":true,"type":"default","frameId":"B7D4FA269922BAFB9A47BB4A8C144B97"}}}
CDP: loadEventFired (waited for)
<- expect_load done
-> wait: ph_is_present("#login")
<- {'type': 'undefined'}
-> wait: ph_is_visible("#login")
<- {'type': 'undefined'}
-> wait: ph_is_present("#login-user-input:not([disabled]):not([aria-disabled=true])")
<- {'type': 'undefined'}
-> wait: ph_is_visible("#login-user-input:not([disabled]):not([aria-disabled=true])")
<- {'type': 'undefined'}
-> ph_set_val("#login-user-input","admin")
<- {'type': 'undefined'}
-> wait: ph_is_present("#login-password-input:not([disabled]):not([aria-disabled=true])")
<- {'type': 'undefined'}
-> wait: ph_is_visible("#login-password-input:not([disabled]):not([aria-disabled=true])")
<- {'type': 'undefined'}
-> ph_set_val("#login-password-input","foobar")
<- {'type': 'undefined'}
-> window.localStorage.setItem("superuser:admin", "any");
<- {'type': 'undefined'}
-> wait: ph_is_present("#login-button:not([disabled]):not([aria-disabled=true])")
<- {'type': 'undefined'}
-> wait: ph_is_visible("#login-button:not([disabled]):not([aria-disabled=true])")
<- {'type': 'undefined'}
-> ph_mouse("#login-button:not([disabled]):not([aria-disabled=true])","click",0,0,0,false,false,false,false)
<- {'type': 'undefined'}
-> expect_load
vCDP: {"source":"network","level":"error","text":"Failed to load resource: the server responded with a status of 500 (Authentication failed: Timeout)","timestamp":1644483196460.968,"url":"http://127.0.0.2:9091/cockpit/login","networkRequestId":"35289.13"}
> warning: fatal: Authentication failed: Timeout
```

It seems like a race condition in the container startup/ssh configuration? the second retry always seems to work.

**Pull request merged**